app/lib/analyzer.dart

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library pub_dartlang_org.analysis;
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:yaml/yaml.dart' as yaml;
+
+import 'model_properties.dart';
+import 'utils.dart';
+
+/// Collects the preparation, analysis and cleanup functions for analyzing a
+/// package.
+class Analyzer {
+  File _sourceTarGz;
+  bool _clearWorkDir = false;
+  Directory _workDir;
+  Directory _packageDir;
+
+  Pubspec pubspec;
+  PubspecLock pubspecLock;
+
+  Analyzer(String source, {String workDir}) {
+    _sourceTarGz = new File(source);
+    if (workDir != null) {
+      _workDir = new Directory(workDir);
+    }
+  }
+
+  /// Prepares the directories for the analysis and extracts the package.
+  Future setUp() async {
+    if (!_sourceTarGz.existsSync()) {
+      throw new Exception('Source file not found: ${_sourceTarGz.path}');
+    }
+    if (_workDir == null) {
+      _workDir = await Directory.systemTemp.createTemp();
+      _clearWorkDir = true;
+    }
+    if (!_workDir.existsSync()) {
+      await _workDir.create(recursive: true);
+    }
+    _packageDir = await _workDir.createTemp('package-');
+    await extractTarball(_sourceTarGz.path, _packageDir.path);
+  }
+
+  Future analyzeAll() async {
+    await runPubGet();
```

Comment:
This code precedes my awareness of `pana`. I think `pana` is a better place for most of this analysis, and I'm planning to contribute to it (e.g. https://github.com/kevmoo/pana.dart/pull/3). At the same time as soon as it is 'good enough', I'll update this PR with the 'pana' version.